plugin/contrib/mesos/pkg/scheduler/service/service_test.go

```diff
@@ -106,3 +113,40 @@ func Test_awaitFailoverDoneFailover(t *testing.T) {
 		t.Fatalf("expected call to failover handler")
 	}
 }
+
+func Test_StaticPods(t *testing.T) {
+	assert := assert.New(t)
+
+	// create static pods config files, spod1 on toplevel and spod2 in a directory "dir"
+	staticPodsConfigPath, err := ioutil.TempDir(os.TempDir(), "executor-k8sm-archive")
+	assert.NoError(err)
+	defer os.RemoveAll(staticPodsConfigPath)
+
+	spod1, err := os.Create(filepath.Join(staticPodsConfigPath, "spod1.json"))
+	assert.NoError(err)
+	spod1.WriteString("content1")
+
+	err = os.Mkdir(filepath.Join(staticPodsConfigPath, "dir"), 0755)
+	assert.NoError(err)
+
+	spod2, err := os.Create(filepath.Join(staticPodsConfigPath, "dir", "spod2.json"))
+	assert.NoError(err)
+	spod2.WriteString("content2")
+
+	// archive config files
+	data, fileNum, err := manifests(staticPodsConfigPath)
+	assert.NoError(err)
+	assert.Equal(2, fileNum)
+
+	// unarchive config files
+	zr, err := zip.NewReader(bytes.NewReader(data), int64(len(data)))
```

Comment:
Since this zip decoding is already tested elsewhere, I think it'd be enough to compare the actual returned byte slice.
```go
if got, want := data, []byte{...}; !bytes.Equal(got, want) {
   t.Fatalf("got: %v, want: %v")
}
```